src/game_battlealgorithm.cpp

```diff
@@ -1354,14 +1400,34 @@ bool Game_BattleAlgorithm::Skill::Execute() {
 				this->sp = std::min<int>(effect, GetTarget()->GetSp());
 			}
 
-			if (skill.affect_attack && Utils::PercentChance(to_hit))
-				this->attack = std::max<int>(0, std::min<int>(effect, GetTarget()->GetAtk() - (GetTarget()->GetBaseAtk() + 1) / 2));
-			if (skill.affect_defense && Utils::PercentChance(to_hit))
-				this->defense = std::max<int>(0, std::min<int>(effect, GetTarget()->GetDef() - (GetTarget()->GetBaseDef() + 1) / 2));
-			if (skill.affect_spirit && Utils::PercentChance(to_hit))
-				this->spirit = std::max<int>(0, std::min<int>(effect, GetTarget()->GetSpi() - (GetTarget()->GetBaseSpi() + 1) / 2));
-			if (skill.affect_agility && Utils::PercentChance(to_hit))
-				this->agility = std::max<int>(0, std::min<int>(effect, GetTarget()->GetAgi() - (GetTarget()->GetBaseAgi() + 1) / 2));
+			if (skill.affect_attack && Utils::PercentChance(to_hit)) {
+				if (!this->healing) {
+					this->attack = std::max<int>(0, std::min<int>(effect, GetTarget()->GetAtk() - (GetTarget()->GetBaseAtk() + 1) / 2));
```

Comment:
Since you're refactoring and testing here, this min/max junk should be replaced with `Utils::Clamp()`

Reply:
Did the replacements now.